// arb-ts/dist/lib/abi/factories/IERC777Sender__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.IERC777Sender__factory = void 0;
var ethers_1 = require("ethers");
var IERC777Sender__factory = /** @class */ (function () {
    function IERC777Sender__factory() {
    }
    IERC777Sender__factory.connect = function (address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    };
    return IERC777Sender__factory;
}());
exports.IERC777Sender__factory = IERC777Sender__factory;
var _abi = [
    {
        inputs: [
            {
                internalType: 'address',
                name: 'operator',
                type: 'address',
            },
            {
                internalType: 'address',
                name: 'from',
                type: 'address',
            },
            {
                internalType: 'address',
                name: 'to',
                type: 'address',
            },
            {
                internalType: 'uint256',
                name: 'amount',
                type: 'uint256',
            },
            {
                internalType: 'bytes',
                name: 'userData',
                type: 'bytes',
            },
            {
                internalType: 'bytes',
                name: 'operatorData',
                type: 'bytes',
            },
        ],
        name: 'tokensToSend',
        outputs: [],
        stateMutability: 'nonpayable',
        type: 'function',
    },
];
